core/tests/protractor_utils/forms.js

```diff
@@ -523,8 +529,15 @@ var RichTextChecker = async function(arrayOfElems, arrayOfTexts, fullText) {
       var elem = await arrayOfElems.get(arrayPointer);
       expect(await elem.getTagName()).
         toBe('oppia-noninteractive-' + componentName.toLowerCase());
-      expect(await elem.getText()).toBe(arrayOfTexts[arrayPointer]);
-
+      // TODO(#9821) :The Math rich text components are now rendered as SVGs by
+      // Math-Jax, and the XPATH_SELECTOR defined in the method expectRichText()
+      // cannot effectively parse the SVG tags, hence the elem.getText() won't
+      // work for Math components.
+      if (componentName === 'Math') {
+        expect(await elem.element(by.tagName('script')).isPresent()).toBe(true);
```

Comment:
Can we make this check more thorough? It's pretty lenient at the moment, which is particularly problematic since in the e2e test file, we'll just see a call to `readRteComponent`, so it'll look thorough even though it's not

Reply:
done, I've added more checks.
But FYI, we are gonna soon change how math rich text components are rendered(we will use images), so I am planning to add more detailed checks after that.